Assets/LeapMotion/Core/Scripts/Utils/XRSupportUtil.cs

```diff
@@ -24,50 +24,69 @@ namespace Leap.Unity {
   public static class XRSupportUtil {
 
     public static bool IsXREnabled() {
-      #if UNITY_5
-      return VRSettings.enabled;
-      #else
+      #if UNITY_2017_2_OR_NEWER
       return XRSettings.enabled;
+      #else
+      return VRSettings.enabled;
       #endif
     }
 
     public static bool IsXRDevicePresent() {
-      #if UNITY_5
+      #if UNITY_2017_2_OR_NEWER
+      return XRDevice.isPresent;
+      #else
       return VRDevice.isPresent;
+      #endif
+    }
+
+    static bool outputPresenceWarning = false;
+    public static bool IsUserPresent() {
+      #if UNITY_2017_2_OR_NEWER
+      var userPresence = XRDevice.userPresence;
+      if (userPresence == UserPresenceState.Present) {
+        return true;
+      } else if (!outputPresenceWarning && userPresence == UserPresenceState.Unsupported) {
+        Debug.LogWarning("XR UserPresenceState unsupported (XR support is probably disabled).");
+        outputPresenceWarning = true;
+      }
       #else
-      return XRDevice.isPresent;
+      if (!outputPresenceWarning){
+        Debug.LogWarning("XR UserPresenceState is only supported in in 2017.2 and newer.");
+        outputPresenceWarning = true;
+      }
       #endif
+      return false;
```

Comment:
Might be worth discussing whether or not this method should assume that 'unsupported' == 'false'.  This might cause certain behaviors to _never_ trigger because they are waiting for user presence.  Maybe defaulting to _true_ might be a good default, possibly with the option to pass in an optional param that specifies the value to return if the feature is unsupported?